lib/ex_plasma/merkle.ex

```diff
@@ -0,0 +1,52 @@
+defmodule ExPlasma.Merkle do
+  @moduledoc """
+  Encapsulates all the interactions with the MerkleTree library.
+  """
+
+  alias ExPlasma.Crypto
+
+  @transaction_merkle_tree_height 16
+  @default_leaf <<0::256>>
+
+  @doc """
+  Generate a Merkle Root hash for the given list of transactions in encoded byte form.
+
+  ## Examples
+
+    iex> txns = %ExPlasma.Transaction{tx_type: 1} |> ExPlasma.encode() |> List.wrap()
+    iex> ExPlasma.Merkle.root_hash(txns)
+    <<168, 54, 172, 201, 1, 212, 18, 167, 34, 57, 232, 89, 151, 225, 172, 150, 208,
+      77, 194, 12, 174, 250, 146, 254, 93, 42, 28, 253, 203, 237, 247, 62>>
+  """
+  @spec root_hash([binary(), ...]) :: binary()
+  def root_hash(encoded_transactions) do
+    MerkleTree.fast_root(encoded_transactions,
+      hash_function: &Crypto.keccak_hash/1,
+      height: @transaction_merkle_tree_height,
+      default_data_block: @default_leaf
+    )
+  end
+
+  # Creates a Merkle proof that transaction under a given transaction index
+  # is included in block consisting of hashed transactions
+  @spec proof([binary(), ...], non_neg_integer()) :: binary()
+  def proof(encoded_transactions, txindex) do
+    encoded_transactions
+    |> build()
+    |> prove(txindex)
+    |> Enum.reverse()
+    |> Enum.join()
+  end
+
+  defp build(encoded_transactions) do
```

Comment:
just a question. this function accepts all transactions at once. is it possible to create a builder that accepts transactions one by one?

Reply:
Yep, we can create a builder for a single transaction but I don't think we have a use for it.

Reply:
it would be useful for block hash generation in childchain. Instead of loading all transactions, we will be able to load the one by one

Reply:
this is - I presume, less related to this chunk of code but if `MerkleTree ` allows lazy encoding.

Reply:
but yeah, this needs to be explored because loading full blocks into memory (64000 transactions) might be less optimal. @ayrat555 can you create a ticket?